lib/segment/src/segment/entry.rs

```diff
@@ -449,14 +449,19 @@ impl SegmentEntry for Segment {
             .map(|data| data.vector_storage.borrow().available_vector_count())
             .sum();
 
-        let vector_data_info = self
+        let mut total_segment_size_bytes = 0;
+
+        let vector_data_info: HashMap<_, _> = self
             .vector_data
             .iter()
             .map(|(key, vector_data)| {
                 let vector_storage = vector_data.vector_storage.borrow();
                 let num_vectors = vector_storage.available_vector_count();
                 let vector_index = vector_data.vector_index.borrow();
                 let is_indexed = vector_index.is_index();
+
+                total_segment_size_bytes += vector_storage.available_size_in_bytes();
```

Comment:
Is `available_size_in_bytes` is an occupied disk volume? It reads as empty space actually.

Also I would prefer to avoid using calls to the file system in this API, as it might have performance impact.

Reply:
yeah the name is confusing but it actually estimates the space occupied by the vector in the segment:

It has many implementations but they are all trying to do the same thing 
```rs
// default trait implementation:
fn available_vector_count(&self) -> usize {
        self.total_vector_count()
            .saturating_sub(self.deleted_vector_count())
}

// for simple_sparse_vector_storage:
fn available_size_in_bytes(&self) -> usize {
        self.total_sparse_size * (std::mem::size_of::<f32>() + std::mem::size_of::<u32>())
}
```

Note that they don't access filesystem so perf will not be impacted.